controller/backup_controller.go

```diff
@@ -160,28 +160,9 @@ func (bc *BackupController) handleErr(err error, key interface{}) {
 		return
 	}
 
-	// The resync period of the backup is one hour and the maxRetries is 3.
-	// Thus, the deletion failure of the backup in error state is caused by the shutdown of the replica during backing up,
-	// if the lock hold by the backup job is not released.
-	// The workaround is to increase the maxRetries number and to retry the deletion until the lock acquisition
-	// of the backup is timeout after 150 seconds.
-	if strings.Contains(err.Error(), "failed lock") {
-		if bc.queue.NumRequeues(key) < maxRetriesOnAcquireLockError {
-			bc.logger.WithError(err).Warnf("Error syncing Longhorn backup %v because of the failure of lock acquisition", key)
-			bc.queue.AddRateLimited(key)
-			return
-		}
-	} else {
-		if bc.queue.NumRequeues(key) < maxRetries {
-			bc.logger.WithError(err).Warnf("Error syncing Longhorn backup %v", key)
-			bc.queue.AddRateLimited(key)
-			return
-		}
-	}
-
-	utilruntime.HandleError(err)
-	bc.logger.WithError(err).Warnf("Dropping Longhorn backup %v out of the queue", key)
-	bc.queue.Forget(key)
+	bc.logger.WithError(err).Warnf("Error syncing Longhorn backup %v", key)
+	bc.queue.AddRateLimited(key)
+	return
```

Comment:
Yes, it is not easy to differentiate recoverable and unrecoverable errors. This can be an improvement ticket and we should apply the improvement to all controllers. 

In the past, we usually tackle this issue by having a resync period. However, we are trying to move away from it thus AddRateLimited is used here

Reply:
Please add this to phase 2 if needed. @PhanLe1010